drivers/dma/dma_stm32u5.c

```diff
@@ -544,11 +544,9 @@ static int dma_stm32_reload(const struct device *dev, uint32_t id,
 				src, dst);
 
 	if (stream->source_periph) {
-		LL_DMA_SetBlkDataLength(dma, dma_stm32_id_to_stream(id),
-				     size / stream->src_size);
+		LL_DMA_SetBlkDataLength(dma, dma_stm32_id_to_stream(id), size);
 	} else {
-		LL_DMA_SetBlkDataLength(dma, dma_stm32_id_to_stream(id),
-				     size / stream->dst_size);
+		LL_DMA_SetBlkDataLength(dma, dma_stm32_id_to_stream(id), size);
 	}
```

Comment:
Identical code in `if` and `else` blocks; check should be removed.

Reply:
I removed the check